src/app/qgslayertreeviewfilterindicator.cpp

```diff
@@ -49,15 +49,24 @@ QString QgsLayerTreeViewFilterIndicatorProvider::iconName( QgsMapLayer *layer )
 QString QgsLayerTreeViewFilterIndicatorProvider::tooltipText( QgsMapLayer *layer )
 {
   QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer );
+  QString filter;
   if ( vlayer )
-    return QStringLiteral( "<b>%1:</b><br>%2" ).arg( tr( "Filter" ), vlayer->subsetString().toHtmlEscaped() );
+    filter = vlayer->subsetString();
 
   // PG raster
   QgsRasterLayer *rlayer = qobject_cast<QgsRasterLayer *>( layer );
   if ( rlayer && rlayer->dataProvider() && rlayer->dataProvider()->supportsSubsetString() )
-    return QStringLiteral( "<b>%1:</b><br>%2" ).arg( tr( "Filter" ), rlayer->subsetString().toHtmlEscaped() );
+    filter = rlayer->subsetString();
 
-  return QString();
+  if ( filter.isEmpty() )
+    return QString();
+
+  if ( filter.size() > 1024 )
+  {
+    filter = filter.left( 1023 ) + QString( QChar( 0x2026 ) );
```

Comment:
@nyalldawson , have you thought about truncating the middle part of the string instead? I.e. `FID IN (1, 2, 3, 4, ... 89, 90)` ?

Reply:
Ok, done

Reply:
@nyalldawson , much appreciated. I love the utils function too, I assume it'll pop up in many places over the next few weeks/months :) )